rdr_service/offline/antibody_study_pipeline.py

```diff
@@ -0,0 +1,257 @@
+
+
+def _import_file(bucket_name, sub_folder_name, file_name_prefix, func):
+    logging.info("Starting antibody study data import from {}.".format(bucket_name))
+    try:
+        file_list = _find_csv_file_list(bucket_name, file_name_prefix, sub_folder_name)
+    except FileNotFoundError:
+        logging.info("File not found")
+        return None
+    if not file_list:
+        logging.info("No file found for importing")
+        return None
+    for (csv_file_cloud_path, csv_filename) in file_list:
+        with open_cloud_file(csv_file_cloud_path) as csv_file:
+            logging.info("Opening  CSV in {}: {}.", bucket_name,
+                         csv_file_cloud_path)
+            csv_reader = csv.DictReader(csv_file, delimiter=",")
+            func(csv_reader, csv_filename)
+            logging.info("import completed, file name: {}".format(csv_filename))
+    logging.info("import all files completed")
+
+
+def _save_biobank_antibody_manifest_from_csv(csv_reader, csv_filename):
+    missing_cols = set(BiobankAntibodyManifestCsvColumns.ALL) - set(csv_reader.fieldnames)
+    if missing_cols:
+        raise DataError("CSV is missing columns %s, had columns %s." % (missing_cols, csv_reader.fieldnames))
+
+    biobank_covid_antibody_dao = BiobankCovidAntibodySampleDao()
+    records = []
+    rows = list(csv_reader)
+    with biobank_covid_antibody_dao.session() as session:
+        for row in rows:
+            if not row[BiobankAntibodyManifestCsvColumns.SAMPLE_ID]:
+                continue
+            record = _create_biobank_covid_antibody_obj_from_row(row, csv_filename)
+            records.append(record)
+            if len(records) >= _BATCH_SIZE:
+                biobank_covid_antibody_dao.upsert_all_with_session(session, records)
+                records = []
+        if records:
+            biobank_covid_antibody_dao.upsert_all_with_session(session, records)
+
+
+def _create_biobank_covid_antibody_obj_from_row(row, csv_filename):
+    biobank_id = row[BiobankAntibodyManifestCsvColumns.BIOBANK_ID]
+    aou_biobank_id = no_aou_biobank_id = None
+
+    if biobank_id and biobank_id.startswith(BIOBANK_ID_PREFIX) and len(biobank_id) == 10:
+        aou_biobank_id = biobank_id[len(BIOBANK_ID_PREFIX):]
+    else:
+        no_aou_biobank_id = biobank_id
+
+    kwargs = dict(
+        aouBiobankId=aou_biobank_id,
+        noAouBiobankId=no_aou_biobank_id,
+        sampleId=row[BiobankAntibodyManifestCsvColumns.SAMPLE_ID],
+        matrixTubeId=row[BiobankAntibodyManifestCsvColumns.MATRIX_ID],
+        sampleType=row[BiobankAntibodyManifestCsvColumns.SAMPLE_TYPE],
+        quantityUl=row[BiobankAntibodyManifestCsvColumns.QUANTITY],
+        storageLocation=row[BiobankAntibodyManifestCsvColumns.STORAGE_LOCATION],
+        collectionDate=row[BiobankAntibodyManifestCsvColumns.COLLECTION_DATE],
+        ingestFileName=csv_filename
+    )
+
+    ojb = BiobankCovidAntibodySample(**kwargs)
+
+    return ojb
```

Comment:
Did you mean to use 'obj' here instead of 'ojb'?

Reply:
Done